### ISSUE TITLE
AssociationProxy and select from an after_create

### ISSUE THREAD

**Issue by maletor** [rails/rails#15258]
Given a `has_many` and an inverse `belongs_to`, when I run `update_all` on the collection from within an `after_create`, ActiveRecord tries to find all records in the association where `id` is `NULL` even though an `id` is present. I expect all associated comments to be updated.

For example,

``` ruby
has_many :comments
after_create do
  comments.select(:id)
  comments.update_all(foo: 'bar')
end
```

will return a `0` even if `comment.size > 0`. If, however, a `reload` is added above `comments.update_all` the `after_create` will run correctly. 

I believe this is a bug with `AssociationProxy` in Rails -- tested on 4.0.5.

**Comment by maletor** [comment]
This issue is stale but still valid.

**Comment by rafaelfranca** [comment]
Thanks

**Comment by ebeigarts** [comment]
This was fixed in #11694. I can reproduce this only with `< 4.0.12`. The test example passes with `4.0.13`, `4.2.x`, `5.0.x`. Please close this.

**Comment by maletor** [comment]
@ebeigarts, thanks.
